### PR TITLE
feat: add C implementation for `math/base/special/beta`

### PR DESCRIPTION
**Pull request by gunjjoshi**
## Description

> What is the purpose of this pull request?

This pull request:

-   adds C implementation for [`math/base/special/beta`](https://github.com/stdlib-js/stdlib/tree/ecf38443532224c35d3f5c00e50a5bdc4801875e/lib/node_modules/%40stdlib/math/base/special/beta).

```c
double stdlib_base_beta( const double a, const double b );
```

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves a part of #649.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

We might not get `100%` code coverage here, due to some lines of code being impossible to reach, such as the condition `x === 0.0` in the `evalrational` function.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by kgryte** [comment]
@gunjjoshi Just checking. Did you compare to the latest Boost implementation? The source in `main.c` says `v1.64`, but I believe Boost has released new versions since then and it would be good to ensure we have the latest source.

**Comment by gunjjoshi** [comment]
@kgryte I looked over the latest Boost implementation, i.e., [`v1.85`](https://www.boost.org/doc/libs/1_85_0/boost/math/special_functions/beta.hpp). After comparing both [`v1.85`](https://www.boost.org/doc/libs/1_85_0/boost/math/special_functions/beta.hpp) and [`v1.64`](https://www.boost.org/doc/libs/1_64_0/boost/math/special_functions/beta.hpp), it seems that there is no difference in their implementations. Shall I just update the version in `main.c` then?

**Comment by kgryte** [comment]
Yes, you can update the version in both the JavaScript and C implementation.

File: lib/node_modules/@stdlib/math/base/special/beta/README.md [hunk 1]
```diff
@@ -113,6 +113,99 @@ for ( x = 0; x < 10; x++ ) {
...
+// returns 1.0
+
+out = stdlib_base_beta( 5.0, 0.2);
+// returns ~3.382
+```
+
+The function accepts the following arguments:
+
+-   **a**: `[in] double` input value.
+-   **b**: `[in] double` power parameter.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/beta/README.md]:
```suggestion
-   **b**: `[in] double` input value.
```

**Review by Planeshifter — APPROVED** [review]
Looks all good to me, thanks @gunjjoshi!

**Pull request merged**